chainer/training/triggers/interval.py

```diff
@@ -40,9 +42,16 @@ def __call__(self, trainer):
         """
         updater = trainer.updater
         if self.unit == 'epoch':
-            prev = self.count
-            self.count = updater.epoch_detail // self.period
-            return prev != self.count
+            epoch_detail = updater.epoch_detail
+            previous_epoch_detail = updater.previous_epoch_detail
```

Comment:
I agree that it is inconvenient to prohibit iterators/updaters without `previous_epoch_detail` because some users already made custom iterators/updaters without it. However, if the iterator/updater does not have `previous_epoch_detail`, there is a risk of resuming issue (#2244). It is danger to ignore the absence of `previous_epoch_detail`.

How about this solution?
If `IntervalTrigger` can not get `previous_epoch_detail`, it fallbacks to previous behaviour. In addition, it raises a warning saying "Without previous_epoch_datail, IntervalTrigger has a resuming issue. Please implement this property to your iterator/updater".